src/pages/en/guides/content-collections.md

```diff
@@ -165,6 +165,28 @@ You can use all of Zod’s properties and methods with content schemas. This inc
 
 📚 See [Zod’s documentation](https://github.com/colinhacks/zod) for a complete list of features.
 
+### Custom entry slugs
+
+By default, Astro will generate a `slug` for each content entry based on its file path. If you want to generate custom slugs for each entry, you can provide a `slug()` function in `defineCollection()`. A custom `slug()` function will be passed the entry ID, default slug, parsed frontmatter (as `data`), and the raw body of the entry.
+
+```ts {5-9}
+// src/content/config.ts
+import { defineCollection } from 'astro:content';
+
+const blog = defineCollection({
+  slug: ({ id, defaultSlug, data, body }) => {
+    // If set, use `permalink` from the entry’s frontmatter as the slug.
```

Comment:
```suggestion
    // Use `permalink` from the entry’s frontmatter as the slug, if it exists.
```
The "if set" threw me off at first. ("If a slug function is set?") I think maybe it's clearer to start with the intention here.

Would also tend to go with "if it exists" or "if found", but that might just be a preference thing.